query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/FilterToPromqlConverter.java

```diff
@@ -0,0 +1,115 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression;
+
+import com.google.protobuf.ByteString;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.commons.codec.binary.Hex;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.Value;
+
+class FilterToPromqlConverter {
+
+  /**
+   * only `AND` operator in filter is allowed
+   *
+   * <p>rhs of leaf filter should be literal
+   */
+  void convertFilterToString(
+      Filter filter,
+      List<String> filterList,
+      String timeFilterColumn,
+      Function<Expression, String> expressionToColumnConverter) {
+    if (filter.getChildFilterCount() > 0) {
+      for (Filter childFilter : filter.getChildFilterList()) {
+        convertFilterToString(
+            childFilter, filterList, timeFilterColumn, expressionToColumnConverter);
+      }
+    } else {
+      if (QueryRequestUtil.isSimpleColumnExpression(filter.getLhs())
+          && timeFilterColumn.equals(
+              getLogicalColumnNameForSimpleColumnExpression(filter.getLhs()))) {
+        return;
+      }
+      StringBuilder builder = new StringBuilder();
+      builder.append(expressionToColumnConverter.apply(filter.getLhs()));
+      builder.append(convertOperatorToString(filter.getOperator()));
+      builder.append(convertLiteralToString(filter.getRhs().getLiteral()));
```

Comment:
What will the null value `convertLiteralToString` mean here? I haven't try such query? Instead of null, would it make to have `""` filter or no filter at all?

Reply:
Throwing exception from convertLiteralToString